### ISSUE TITLE
[Feature Request] GCP Cloud Storage History Backend

### ISSUE THREAD

**Issue by GoodmanBen** [minamijoyo/tfmigrate#72]
Current State:
At present, `tfmigrate` only supports remote storage in Amazon's S3 service. This limits the usage for projects that are based on GCP.

Desired New State:
Support GCP's Cloud Storage as a remote storage service.

**Comment by GoodmanBen** [comment]
Will follow the pattern recommended by [this comment](https://github.com/minamijoyo/tfmigrate/issues/21#issuecomment-822300905). Assigning this issue to myself, @GoodmanBen.

**Comment by minamijoyo** [comment]
Hi @GoodmanBen, Thank you for offering to take over #23.

As mentioned in https://github.com/minamijoyo/tfmigrate/pull/23#discussion_r697307700, the patch has a license conflict issue and I've been waiting a response from @ethernetdan.

With no reply for months, if you are willing to help with us, it seems that it's time to move forward with you. If I split the history storage related implementation into another repository as MPL2, would you implement the GCC history storage?

**Comment by GoodmanBen** [comment]
@minamijoyo Yes, happy to do so in another repository if that is where you want it to live.

**Comment by minamijoyo** [comment]
Hi @GoodmanBen,

Splitting the history storage related implementation into a new separate repository in MPL2.0 has been done by #79, #80.
https://github.com/minamijoyo/tfmigrate-storage

Feel free to reuse some code snippet from the upstream hashicorp/terraform to implement the gcs storage support.

**Comment by GoodmanBen** [comment]
Sounds great, thank you for doing this refactor.
